compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -32,6 +32,105 @@ enum Position {
     Named(Symbol, InnerSpan),
 }
 
+/// Indicates how positional named argument (i.e. an named argument which is used by position
+/// instead of by name) is used in format string
+/// * `Arg` is the actual argument to print
+/// * `Width` is width format argument
+/// * `Precision` is precion format argument
+/// Example: `{Arg:Width$.Precision$}
+#[derive(Debug, Eq, PartialEq)]
+enum PositionalNamedArgType {
+    Arg,
+    Width,
+    Precision,
+}
+
+/// Contains information necessary to create a lint for a positional named argument
+#[derive(Debug)]
+struct PositionalNamedArg {
+    ty: PositionalNamedArgType,
+    /// The piece of the using this argument (multiple pieces can use the same argument)
+    cur_piece: usize,
+    /// The InnerSpan for in the string to be replaced with the named argument
+    /// This will be None when the position is implicit
+    inner_span_to_replace: Option<rustc_parse_format::InnerSpan>,
+    /// The name to use instead of the position
+    replacement: Symbol,
+    /// The span for the positional named argument (so the lint can point a message to it)
+    positional_named_arg_span: Span,
+}
+
+impl PositionalNamedArg {
+    /// Determines what span to replace with the name of the named argument
+    fn get_span_to_replace(&self, cx: &Context<'_, '_>) -> Option<Span> {
+        if let Some(inner_span) = &self.inner_span_to_replace {
+            return match self.ty {
+                PositionalNamedArgType::Arg | PositionalNamedArgType::Width => Some(Span::new(
+                    cx.fmtsp.lo() + BytePos(inner_span.start.try_into().unwrap()),
+                    cx.fmtsp.lo() + BytePos(inner_span.end.try_into().unwrap()),
+                    self.positional_named_arg_span.ctxt(),
+                    self.positional_named_arg_span.parent(),
+                )),
+                PositionalNamedArgType::Precision => Some(Span::new(
```

Comment:
What's the reason for `+ BytePos(1)`? Leave a comment plz. Or better yet, it would be nice to adjust the place you're getting this from.

And it would also be nice if we could use `Span::from_inner` here too.

Reply:
Changed so the increment is done in `maybe_push` and added a big comment explaining why. Please let me know if comment is unclear.

And used `from_inner`!